ansible/library/port_alias.py

```diff
@@ -101,33 +123,64 @@ def get_portmap(self):
                             alias_index = index
                         if 'speed' in text:
                             speed_index = index
+                        if 'role' in text:
+                            role_index = index
             else:
                 if re.match('^Ethernet', line):
                     mapping = line.split()
                     name = mapping[0]
+                    if (role_index != -1) and (len(mapping) > role_index):
+                        role = mapping[role_index]
+                    else:
+                        role = 'Ext'
                     if alias_index != -1 and len(mapping) > alias_index:
                         alias = mapping[alias_index]
                     else:
                         alias = name
-                    aliases.append(alias)
-                    portmap[name] = alias
-                    aliasmap[alias] = name
-                    if (speed_index != -1) and (len(mapping) > speed_index):
-                        portspeed[alias] = mapping[speed_index]
+                    if (all_ports == 'False' and role == 'Ext') or \
+                       (all_ports == 'True'):
+                        aliases.append(alias)
+                        portmap[name] = alias
+                        aliasmap[alias] = name
+                        if (speed_index != -1) and (len(mapping) > speed_index):
+                            portspeed[alias] = mapping[speed_index]
 
         return (aliases, portmap, aliasmap, portspeed)
 
 def main():
     module = AnsibleModule(
         argument_spec=dict(
-            hwsku=dict(required=True, type='str')
+            hwsku=dict(required=True, type='str'),
+            num_asic=dict(type='int', required=False),
+            all_ports=dict(type='str', default='True', required=False)
         ),
         supports_check_mode=True
     )
     m_args = module.params
     try:
+        aliases = []
+        portmap = {}
+        aliasmap = {}
+        portspeed = {}
         allmap = SonicPortAliasMap(m_args['hwsku'])
-        (aliases, portmap, aliasmap, portspeed) = allmap.get_portmap()
+        all_ports = m_args['all_ports']
+        if m_args['num_asic'] is not None:
+            num_asic = m_args['num_asic']
+        else:
+            num_asic = allmap.get_num_asic()
```

Comment:
use sonic-py-common utility here

Reply:
Modified to use sonic_py_common utility.
But as port_alias.py script is invoked from sonic_mgmt docker during VS bring up or is run on the device/VS during minigraph generation, the import of sonic_py_common library cannot be done globally as it will not present when run on sonic-mgmt docker.